src/ray/common/ray_syncer/common.h

```diff
@@ -16,11 +16,18 @@
 
 #pragma once
 
+#include <functional>
+
 #include "src/ray/protobuf/ray_syncer.grpc.pb.h"
+#include "src/ray/protobuf/ray_syncer.pb.h"
 
 namespace ray::syncer {
 
 inline constexpr size_t kComponentArraySize =
     static_cast<size_t>(ray::rpc::syncer::MessageType_ARRAYSIZE);
 
+// Observer callable for sync message response.
+using RaySyncMsgObserver =
+    std::function<void(const ::ray::rpc::syncer::RaySyncMessage &)>;
```

Comment:
Can we instead define a generic

```
// A callback is called whenver a rpc completes between the current process and the given raylet regardless of whether the response indicates success or failure
using RayletCompletedRpcCallback(void(const NodeID &))
```

and this callback can be set on any components that have rpcs with raylet.

Reply:
> regardless of whether the response indicates success or failure

I'm not sure this is correct. For example, if you get an error status on grpc (i.e. UNAVAILABLE for grpc status), it actually means communication between gcs and raylet is broken, which we shouldn't record completion.

Reply:
Here `Completed` means `client sends the request and then receives the response` so if this is UNAVAILABLE, it's not completed. This is the naming suggested by chatgpt lol.

Reply:
Or we can think `Completed` == GrpcStatus::OK

Reply:
But feel free to give a better name.

Reply:
Updated.

Reply:
I removed `Raylet` from the comment and type alias, the whole `RaySyncer` is pretty general-purpose, which doesn't couple with any component in terms of implementation.